### PR TITLE
Fix required_extension behavior in read_parquet

### LINKED ISSUE #8349: Regression: `read_parquet` fails on main (not on 2021.10.0)
There is a regression on main that causes the following to fail. I discovered this while looking at https://github.com/dask/dask/issues/8323

```python
import numpy as np
import pandas as pd
import dask.dataframe as dd

np.random.seed(1)
df = pd.DataFrame(np.random.randint(0, 10, size=(10, 4)), columns=['a', 'b', 'c', 'd']).set_index(['a', 'b']).sort_index()
df.to_parquet("multi.pq", engine='fastparquet')
dd.read_parquet("multi.pq").compute()
```

```python-traceback
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-4-79a798c099ae> in <module>
      6 df = pd.DataFrame(np.random.randint(0, 10, size=(10, 4)), columns=['a', 'b', 'c', 'd']).set_index(['a', 'b']).sort_index()
      7 df.to_parquet("multi.pq", engine='fastparquet')
----> 8 dd.read_parquet("multi.pq").compute()

~/dask/dask/dataframe/io/parquet/core.py in read_parquet(path, columns, filters, categories, index, storage_options, engine, gather_statistics, ignore_metadata_file, metadata_task_size, split_row_groups, chunksize, aggregate_files, **kwargs)
    324         gather_statistics = True
    325 
--> 326     read_metadata_result = engine.read_metadata(
    327         fs,
    328         paths,

~/dask/dask/dataframe/io/parquet/fastparquet.py in read_metadata(cls, fs, paths, categories, index, gather_statistics, filters, split_row_groups, chunksize, aggregate_files, ignore_metadata_file, metadata_task_size, **kwargs)
    849 
    850         # Stage 1: Collect general dataset information
--> 851         dataset_info = cls._collect_dataset_info(
    852             paths,
    853             fs,

~/dask/dask/dataframe/io/parquet/fastparquet.py in _collect_dataset_info(cls, paths, fs, categories, index, gather_statistics, filters, split_row_groups, chunksize, aggregate_files, ignore_metadata_file, metadata_task_size, require_extension, **kwargs)
    455                 # Will need to pass a list of paths to read_partition
    456                 scheme = get_file_scheme(fns)
--> 457                 pf = ParquetFile(paths[:1], open_with=fs.open, root=base, **kwargs)
    458                 pf.file_scheme = scheme
    459                 pf.cats = paths_to_cats(fns, scheme)

~/conda/envs/dask-dev/lib/python3.9/site-packages/fastparquet/api.py in __init__(self, fn, verify, open_with, root, sep, fs, pandas_nulls)
    103             fs = getattr(open_with, "__self__", None)
    104         if isinstance(fn, (tuple, list)):
--> 105             basepath, fmd = metadata_from_many(fn, verify_schema=verify,
    106                                                open_with=open_with, root=root,
    107                                                fs=fs)

~/conda/envs/dask-dev/lib/python3.9/site-packages/fastparquet/util.py in metadata_from_many(file_list, verify_schema, open_with, root, fs)
    177                     raise ValueError('Incompatible schemas')
    178 
--> 179         fmd = copy.copy(pfs[0].fmd)  # we inherit "created by" field
    180         fmd.row_groups = []
    181 

IndexError: list index out of range
```

ping @martindurant and @rjzamora

### PR DESCRIPTION
**Pull request by rjzamora**
This PR changes the behavior of `read_parquet(path, ..., required_extension=)` to only apply when `path` is a directory without an active `_metadata` file. That is, we should not be filtering out "unsupported extensions" when the `path` is a single file, and specific list of files, or when we are using a `_metadata` file to tell us what files are valid.

- [x] Closes #8349
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by jakirkham** [comment]
cc @jsignell @martindurant @jrbourbeau

File: dask/dataframe/io/parquet/arrow.py
Comment (jsignell) [inline dask/dataframe/io/parquet/arrow.py:817]: I think this error would be slightly more informative if it included the value of `require_extension`
```suggestion
                        "No files satisfy the `required_extension` criteria "
                        f"(files must end with {required_extension})."
```

**Review by jsignell — APPROVED** [review]
This looks nice! Thanks for the speedy turn around. I just have one comment about making the error a bit more informative.

File: dask/dataframe/io/parquet/fastparquet.py
Comment (rjzamora) [reply dask/dataframe/io/parquet/fastparquet.py:442]: Yup!

**Review by jrbourbeau — COMMENTED** [review]
Thanks for the quick fix @rjzamora! I'll hold off on releasing until this is merged

**Pull request merged**
